editor/src/components/navigator/navigator-item/navigator-item-dnd-container.tsx

```diff
@@ -111,6 +111,65 @@ export interface ConditionalClauseNavigatorItemContainerProps
   navigatorEntry: ConditionalClauseNavigatorEntry
 }
 
+function isDroppingToOriginalPosition(
+  metadata: ElementInstanceMetadataMap,
+  elementPathTree: ElementPathTreeRoot,
+  dropTargetHint: DropTargetHint,
+  dropTargetElementPath: ElementPath,
+  draggedElementPath: ElementPath,
+): boolean {
+  if (EP.pathsEqual(dropTargetElementPath, draggedElementPath)) {
+    return true
+  }
+
+  const parentMatches = EP.pathsEqual(
+    EP.parentPath(draggedElementPath),
+    dropTargetHint.targetParent.elementPath,
+  )
+  if (!parentMatches) {
+    // bail out if not dropping into the same parent
+    return false
+  }
+
+  // dropping the element before itself
+  const index = MetadataUtils.getIndexInParent(metadata, elementPathTree, draggedElementPath)
+  if (
+    dropTargetHint.targetIndexPosition.type === 'before' &&
+    dropTargetHint.targetIndexPosition.index === index
+  ) {
+    return true
+  }
+
+  // dropping the element after the preceding entry
+  if (
+    dropTargetHint.targetIndexPosition.type === 'after' &&
+    dropTargetHint.targetIndexPosition.index === index - 1
+  ) {
+    return true
+  }
+
+  // dropping to front and back
+  const siblings = MetadataUtils.getSiblingsOrdered(metadata, elementPathTree, draggedElementPath)
+  if (
+    (dropTargetHint.targetIndexPosition.type === 'back' &&
+      EP.pathsEqual(siblings.at(0)?.elementPath ?? null, draggedElementPath)) ||
+    (dropTargetHint.targetIndexPosition.type === 'front' &&
+      EP.pathsEqual(siblings.at(-1)?.elementPath ?? null, draggedElementPath))
+  ) {
+    return true
+  }
+
+  // absolute, for the sake of completeness
```

Comment:
you mention completeness, maybe we should assertNever on the indexPosition.type? or is that very _very_ unlikely that we introduce a new indexposition type?